core/src/main/scala/org/apache/spark/resource/ResourceProfileBuilder.scala

```diff
@@ -31,7 +31,7 @@ import org.apache.spark.annotation.Evolving
  * requirements between stages.
  */
 @Evolving
```

Comment:
nit: this mixes two concepts. `Evolving` says it's an API but it's actually private.

Reply:
this is only private until the feature is completely in. I made this such that everything is ready except remove the private once the other prs that make it functional are in. 
I do not want to remove Evolving now because it could be forgotten later.